OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs

```diff
@@ -39,7 +39,7 @@ public void Produce(string actorType, string raceVariant = null)
 			if (!Self.World.Map.Rules.Actors.TryGetValue(actorType, out actorInfo))
 				throw new LuaException("Unknown actor type '{0}'".F(actorType));
 
-			Self.QueueActivity(new WaitFor(() => p.Produce(Self, actorInfo, raceVariant)));
+			Self.QueueActivity(new WaitFor(() => p.Produce(Self, new[] { actorInfo }, raceVariant)));
```

Comment:
This will create a new array each time WaitFor ticks.  Please extract this out into a temporary variable outside the closure.